ambry-protocol/src/main/java/com.github.ambry.protocol/ReplicationControlAdminRequest.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.protocol;
+
+import com.github.ambry.utils.Utils;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * An admin request used to control replication behavior (enable/disable)
```

Comment:
I can add details here if you think it is useful but the constructor and other docs of this class has answers for all these questions. I did not want to repeat it in two places.

Replication disabling is only "from" datacenters i.e. the node will not replicate from the datacenter. The other way round is already supported via the `RequestControlAdminRequest` through which replication metadata requests can be controlled.

We no longer have the concept on intra v/s inter in`ReplicaThread` (except for metrics and the artificial delay). `ReplicaThread` instances are for datacenters. So you can disable replication from the local dc if intra has to be stopped. 

There is no option to specify nodes - only datacenters and partitions. There is no use case right now to do it by node. That support can be added later if required.